jenkins2/Dockerfile

```diff
@@ -0,0 +1,115 @@
+FROM registry.service.opg.digital/opguk/jre-8
+
+#Install tools
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add - && \
+    add-apt-repository \
+      "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+      $(lsb_release -cs) \
+      stable" && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+      apparmor \
+      apt-transport-https \
+      ca-certificates \
+      curl \
+      docker-ce \
+      fontconfig \
+      gettext \
+      libcurl4-gnutls-dev \
+      libexpat1-dev \
+      libffi-dev \
+      libgmp3-dev \
+      libssl-dev \
+      libssl-dev \
+      libx11-6 \
+      libxml2-dev \
+      libxslt1-dev \
+      libyaml-dev \
+      python-dev \
+      python2.7-dev \
+      python3-dev \
+      software-properties-common \
+      ttf-dejavu \
+      zip && \
+    apt-get clean && rm -rf /tmp/* /var/tmp/* && rm -rf /var/lib/apt/lists/* && \
+    pip install -U pip && \
+    pip install \
+      docker-compose virtualenv git+https://github.com/ministryofjustice/semvertag.git
+
+
+### This section is from https://github.com/jenkinsci/docker/blob/master/Dockerfile with changes ###
+
+ENV JENKINS_HOME /srv/jenkins
+ENV JENKINS_SLAVE_AGENT_PORT 50000
+
+ARG user=app
+ARG group=app
+ARG uid=1000
+ARG gid=1000
+
+#Fix docker GID to sync with underlying ECS so that the user can access the docker socket
+###RUN groupmod -g 497 docker && adduser ${user} docker
+
+# Jenkins is run with user `jenkins`, uid = 1000
+# If you bind mount a volume from the host or a data container,
+# ensure you use the same uid
+RUN groupmod -g ${gid} -n ${group} app \
+    && usermod -d "$JENKINS_HOME" -u ${uid} -g ${gid} -m -s /bin/bash -l ${user} app
+
+# Jenkins home directory is a volume, so configuration and build history
+# can be persisted and survive image upgrades
+VOLUME $JENKINS_HOME
+
+# `/usr/share/jenkins/ref/` contains all reference configuration we want
+# to set on a fresh new installation. Use it to bundle additional plugins
+# or config file with your custom jenkins Docker image.
+RUN mkdir -p /usr/share/jenkins/ref/init.groovy.d
+
+COPY init.groovy /usr/share/jenkins/ref/init.groovy.d/tcp-slave-agent-port.groovy
+
+# Bootstrap commands, and our Jenkins service.
+COPY docker/etc/service /etc/sv/
+RUN ln -s /etc/sv/jenkins /etc/service
+
+
+# jenkins version being bundled in this docker image
+ARG JENKINS_VERSION
+ENV JENKINS_VERSION ${JENKINS_VERSION:-2.46.2}
+
+# jenkins.war checksum, download will be validated using it
+ARG JENKINS_SHA=aa7f243a4c84d3d6cfb99a218950b8f7b926af7aa2570b0e1707279d464472c7
```

Comment:
Can you not pull this dynamically? If you use a different Jenkins version this will change